geode-assembly/src/acceptanceTest/java/org/apache/geode/cache/persistence/MissingDiskStoreAcceptanceTest.java

```diff
@@ -70,40 +70,60 @@ public void setUp() throws Exception {
     server1Folder = temporaryFolder.newFolder(SERVER_1_NAME).toPath().toAbsolutePath();
     server2Folder = temporaryFolder.newFolder(SERVER_2_NAME).toPath().toAbsolutePath();
 
-    int[] ports = getRandomAvailableTCPPorts(3);
+    int[] ports = getRandomAvailableTCPPorts(9);
     locatorPort = ports[0];
     int server1Port = ports[1];
     int server2Port = ports[2];
+    int httpPort1 = ports[3];
+    int httpPort2 = ports[4];
+    int httpPort3 = ports[5];
+    int jmxPort1 = ports[6];
+    int jmxPort2 = ports[7];
+    int jmxPort3 = ports[8];
 
     String startLocatorCommand = String.join(" ",
         "start locator",
         "--name=" + LOCATOR_NAME,
         "--dir=" + locatorFolder,
         "--port=" + locatorPort,
+        "--http-service-port=" + httpPort1,
+        "--J=-Dgemfire.jmx-manager-port=" + jmxPort1,
         "--locators=localhost[" + locatorPort + "]");
 
     startServer1Command = String.join(" ",
         "start server",
         "--name=" + SERVER_1_NAME,
         "--dir=" + server1Folder,
         "--locators=localhost[" + locatorPort + "]",
+        "--http-service-port=" + httpPort2,
+        "--J=-Dgemfire.jmx-manager-port=" + jmxPort2,
         "--server-port=" + server1Port);
 
     startServer2Command = String.join(" ",
         "start server",
         "--name=" + SERVER_2_NAME,
         "--dir=" + server2Folder,
         "--locators=localhost[" + locatorPort + "]",
+        "--http-service-port=" + httpPort3,
+        "--J=-Dgemfire.jmx-manager-port=" + jmxPort3,
         "--server-port=" + server2Port);
 
     String createRegionCommand = String.join(" ",
         "create region",
         "--name=" + REGION_NAME,
         "--type=REPLICATE_PERSISTENT");
 
-    gfshRule.execute(startLocatorCommand, startServer1Command, startServer2Command,
-        createRegionCommand);
+    gfshRule.execute(startLocatorCommand);
 
+    gfshRule.execute(startServer1Command, startServer2Command);
+    Thread.sleep(10000);
```

Comment:
Does this sleep still need to be there? It was there when we were trying to sort out the issue, but it is it required?

Reply:
Would be better to `await` until we can assert that the server is started. Arbitrary sleeps make for slower and flakier tests.